### ISSUE TITLE
IfStatementAssignment False Positives on Function Signature

### ISSUE THREAD

**Issue by drmmr763** [phpmd/phpmd#678]
- PHPMD version: 2.7.0
- PHP Version: PHP 7.2.19
- Installation type: composer
- Operating System / Distribution & Version: (Windows 10 / CentOS 7.6 / ...): Ubuntu 

### Current Behavior
Rule `ifstatementassignment` is thrown on a function signature that does not contain any if-statement assignments.

Full Function where Rule is Thrown:
```
public function setOrganizations() : AssociateRecords
    {
        if ($this->request->has('organizations')) {
            // get the user's current organizations so we can properly set updated user / date
            $currentOrganizations = $this->user->organizations->keyBy('id');
            $organizationIDs      = Organization::findMany($this->request->get('organizations'))
                                    ->pluck('id');
            $toSync               = collect();

            // loop over each ID and update or modify the record
            $organizationIDs->each(function ($id) use ($currentOrganizations, $toSync) {
                $createdAt = new \DateTime;
                $createdBy = auth()->id();

                if ($currentOrganizations->has($id)) {
                    /** @var Organization $org */
                    $org       = $currentOrganizations->get($id);
                    $createdAt = $org->pivot->created_at;
                    $createdBy = $org->pivot->created_by;
                }

                $toSync->put($id, [
                    'created_at'  => $createdAt,
                    'created_by'  => $createdBy,
                    'updated_at'  => new \DateTime(),
                    'modified_by' => auth()->id(),
                ]);
            });

            // sync
            $this->user->organizations()->sync($toSync->toArray());
        }

        return $this;
    }
```

### Expected Behavior
The rule should not be thrown in the example code above. I'm not sure exactly what is causing it to trigger. I thought it was the type hinted return type but removing that didn't resolve it. 

### Steps To Reproduce:
Explain **all** the steps you did to create this bug so we can reproduce it. 

* I have been using phpmd on my code base for quite a while and not encountered this violation before
* my code base has not changed significantly and these lines are not newly created. I'm not sure why this just started happening, it popped up in a CI / CD build recently.
* I updated to the latest version of phpmd 2.7 in hopes that it would be resolved, but that did not help. 

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

![image](https://user-images.githubusercontent.com/643104/66411964-a274c680-e9c2-11e9-94dc-8fd634f76d1c.png)

**Comment by kylekatarnls** [comment]
Hi, this is fixed and will be released in 2.7.1. See #661

Thanks.

**Comment by drmmr763** [comment]
Thanks! Sorry for the duplicate. I saw those but thought they were fixed in 2.7.0.

**Comment by kylekatarnls** [comment]
Try to require `"phpmd/phpmd": "dev-master"`, if you have your bug with it, I would re-open the issue.
